pkg/kubernetes/pods.go

```diff
@@ -63,6 +64,41 @@ func GetEnvVarIndex(envVarName string, env *[]corev1.EnvVar) int {
 	return -1
 }
 
+func IsInitContainersEqual(srcContainer, destContainer []corev1.Container) bool {
+	if len(srcContainer) != len(destContainer) {
+		return false
+	}
+	for _, srcInitContainer := range srcContainer {
+		if dstInitContainerIdx := FindInitContainer(destContainer, srcInitContainer.Name); dstInitContainerIdx < 0 {
+			return false
+		} else {
+			if !reflect.DeepEqual(srcInitContainer.Command, destContainer[dstInitContainerIdx].Command) ||
+				!reflect.DeepEqual(srcInitContainer.Args, destContainer[dstInitContainerIdx].Args) {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func IsInitContainerTerminated(initContainerStatuses []corev1.ContainerStatus) bool {
```

Comment:
I think `InitContainerFailed` is more descriptive as we're checking that a container has terminated and with a non-zero exit code.

Reply:
Yes, sure, will fix.